src/main/java/com/nyxei/pollingus/service/QuestionExtractor.java

```diff
@@ -0,0 +1,40 @@
+package com.nyxei.pollingus.service;
+
+import org.commonmark.node.AbstractVisitor;
+import org.commonmark.node.Heading;
+import org.commonmark.node.Paragraph;
+
+import javax.swing.text.ParagraphView;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QuestionExtractor {
+    static class QuestionVisitor extends AbstractVisitor {
+        private List<String> questions;
+
+        public QuestionVisitor(){
+            //initialser la liste
+            questions = new ArrayList<>();

```

Comment:
Architecture flag there.
The `QuestionVisitor` class is the only one to initialise the questions list, but it can't neither modify the list nor give write access to it. _Is it done on purpose ?_
`QuestionVisitor` is unable to operate with a already-created questions list. _It is done on purpose ?_

Reply:
j'ai ajouté une son contructeur qui prend une liste de questions en paramètre